### PR TITLE
Reland "Add a FlutterDriver screenshot test to device lab. #45411"

### PR DESCRIPTION
**Pull request by cyanglaz**
## Description

Reland "Add a FlutterDriver screenshot test to device lab. #45411" while only enable the test on one Mac-14
This solves the framework part of https://github.com/flutter/flutter/issues/45348.

## Related Issues

https://github.com/flutter/flutter/issues/46622

## Tests

I added the following tests:

flutter_driver_screenshot_test_ios on device lab.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: dev/devicelab/manifest.yaml
Comment (cyanglaz) [inline dev/devicelab/manifest.yaml]: Added a new capability to make the test only run on one specific ios device

Comment (digiter) [reply dev/devicelab/manifest.yaml]: Change to `["mac/ios", "ios/gl-render-image"]`, (make sure to quote each term).

Make the description be more specific, explain it makes sure flutter renders an fixed image on ios GL.

Comment (cyanglaz) [reply dev/devicelab/manifest.yaml]: Thanks! I updated the PR.

**Review by digiter — APPROVED** [review]
The dev/devicelab/manifest.yaml LGTM. I didn't review the rest of it because this is a reland.

Defer to @cyanglaz to decide whether this needs another reviewer.

**Comment by cyanglaz** [comment]
@digiter Thanks! The only change in this PR besides the re-land is the `manifest` and golden update. I think we can land it on green.

**Pull request merged**
